samples/sample-manifest.json

```
@@ -20,13 +20,24 @@
             "MaxActivations": 1,
             "Parameters": {
                 "Files": {
-                    "/tmp/defg.txt": "foo",
-                    "/tmp/jkl.mno": "bar"
+                    "/tmp/abc.txt": "helloworld",
+                    "/tmp/helloworld.sh": "#!/usr/bin/env bash\necho {{ raw .Secrets.hello }} {{ raw .Secrets.hello }}",
+                    "/tmp/defg.txt": {
+                        "encoding": "string",
+                        "data": "foo"
+                    },
+                    "/tmp/jkl.mno": {
+                        "encoding": "base64",
+                        "data": "YmFy"
+                    },
+                    "/tmp/enc.key": {
+                        "encoding": "bytes",
+                        "data": "{{ raw .Secrets.symmetric_key_shared }}"
                 },
                 "Env": {
                     "IS_FIRST": "true",
```

```suggestion
                    "IS_FIRST": "true",
                    "IS_SECOND": {
                        "encoding": "string",
                        "data": "false"
                    },
```
Will this be allowed?

---

By default, this is not allowed I think (at least what I tried out yesterday). Maybe with a custom unmarshaler...

---

Would opt not to. Environment variables are strings anyway, and environment != files, so don't think we should actually use a wrapper type + extra logic to handle both cases when it's always supposed to be a string anyway.

---

We decided that we should have consistency in the implementation of 'files' and `env`.
We're not going to advertise that you can use env this way though.

---

Okay, as long as we explicitly disallow any types which might contain NULL bytes (see discussion above), I am fine with that.